examples/other/hirez_screenshot.py

```diff
@@ -0,0 +1,232 @@
+    let r3_out_blue = -r3_out - inner_offset;
+    let r3_in_blue = r3_out_blue + pygfx_width;
+    let r3_blue = max(
+        max(r1_out, r2_in),
+        max(r3_out_blue, -r3_in_blue)
+    );
+
+    let inner_triangle = min(r1_blue, min(r2_blue, r3_blue));
+
+    let outer_triangle = max(
+        max(r1_out, max(r2_out, r3_out)),
+        -max(r1_in, max(r2_in, r3_in))
+    );
+
+    return min(inner_triangle, outer_triangle) * size;
+"""
+
+
+y = 0
+text = gfx.Text(
+    gfx.TextGeometry("centered", anchor="middle-middle", font_size=1),
+    gfx.TextMaterial("#000"),
+)
+text.local.y = y
+text.local.x = npoints
+scene.add(text)
+
+text = gfx.Text(
+    gfx.TextGeometry("inner", anchor="middle-middle", font_size=1),
+    gfx.TextMaterial("#000"),
+)
+text.local.y = y
+text.local.x = 2 * npoints + npoints
+scene.add(text)
+
+text = gfx.Text(
+    gfx.TextGeometry("outer", anchor="middle-middle", font_size=1),
+    gfx.TextMaterial("#000"),
+)
+text.local.y = y
+text.local.x = 4 * npoints + npoints
+scene.add(text)
+
+all_lines = []
+for marker in gfx.MarkerShape:
+    y += 2
+    line = gfx.Points(
+        geometry,
+        gfx.PointsMarkerMaterial(
+            size=1,
+            size_space="world",
+            color_mode="vertex",
+            # color_mode='debug',
+            marker=marker,
+            edge_color="#000",
+            edge_width=0.1 if not marker == "custom" else 0.033333,
+            custom_sdf=pygfx_sdf if marker == "custom" else None,
+        ),
+    )
+    line.local.y = -y
+    line.local.x = 1
+    scene.add(line)
+    all_lines.append(line)
+
+    line_inner = gfx.Points(
+        geometry,
+        gfx.PointsMarkerMaterial(
+            size=1,
+            size_space="world",
+            color_mode="vertex",
+            marker=marker,
+            edge_color="#000",
+            edge_width=0.1 if not marker == "custom" else 0.033333,
+            edge_mode="inner",
+            custom_sdf=pygfx_sdf if marker == "custom" else None,
+        ),
+    )
+
+    line_inner.local.y = -y
+    line_inner.local.x = 1 + 2 * npoints
+
+    scene.add(line_inner)
+    all_lines.append(line_inner)
+
+    line_outer = gfx.Points(
+        geometry,
+        gfx.PointsMarkerMaterial(
+            size=1,
+            size_space="world",
+            color_mode="vertex",
+            marker=marker,
+            edge_color="#000",
+            edge_width=0.1 if not marker == "custom" else 0.033333,
+            edge_mode="outer",
+            custom_sdf=pygfx_sdf if marker == "custom" else None,
+        ),
+    )
+
+    line_outer.local.y = -y
+    line_outer.local.x = 1 + 4 * npoints
+
+    scene.add(line_outer)
+    all_lines.append(line_outer)
+
+    text = gfx.Text(
+        gfx.TextGeometry(marker, anchor="middle-right", font_size=1),
+        gfx.TextMaterial("#000"),
+    )
+    text.local.y = -y
+    text.local.x = 0
+    scene.add(text)
+
+
+camera = gfx.OrthographicCamera()
+camera.show_object(scene, scale=0.7)
```

Comment:
```suggestion
camera.show_object(scene, scale=0.7)

camera_state = camera.get_state()

# To help visualize the rendering ability, we draw two boxes
# One entirely contained within the FOV of the main image, the other
# entirely outside of the FOV of the main image.
# By disabling AA, we should be able to use this to validate that we don't
# render more than we should.
line_thickness = 1
box_width = camera_state['width']
box_height = camera_state['height']
logical_size = canvas.get_logical_size()
if logical_size[0] > logical_size[1]:
    box_width *= logical_size[0] / logical_size[1]
else:
    box_height *= logical_size[1] / logical_size[0]
inner_lines_geometry = gfx.box_geometry(
    width=box_width - line_thickness,
    height=box_height - line_thickness,
)


inner_lines_geometry.positions.data[..., 0] += camera_state['position'][0]
inner_lines_geometry.positions.data[..., 1] += camera_state['position'][1]

outer_lines_geometry = gfx.Geometry(positions=[
    [0, 0, 0],
    [box_width + line_thickness, 0, 0],
    [box_width + line_thickness, box_height + line_thickness, 0],
    [0, box_height + line_thickness, 0],
    [0, 0, 0],
])
outer_lines_geometry.positions.data[..., 0] -= (box_width + line_thickness) / 2
outer_lines_geometry.positions.data[..., 1] -= (box_height + line_thickness) / 2

outer_lines_geometry.positions.data[..., 0] += camera_state['position'][0]
outer_lines_geometry.positions.data[..., 1] += camera_state['position'][1]
scene.add(
    gfx.Line(
        inner_lines_geometry,
        gfx.LineMaterial(
            color="red",
            thickness=line_thickness,
            thickness_space="world",
            aa=False
        ),
    )
)
scene.add(
    gfx.Line(
        outer_lines_geometry,
        gfx.LineMaterial(
            color="blue",
            thickness=line_thickness,
            thickness_space="world",
            aa=False
        ),
    )
)
```

Reply:
I think you should be able to get non integer scaling (i believe in you!)

In my naive implementation this clearly helps visualize the issues:
![image](https://github.com/user-attachments/assets/0e027b02-e2aa-40aa-a05f-3cf8b95d5540)